prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2484,6 +2479,66 @@ int rxr_ep_alloc_app_device_info(struct fi_info **app_device_info,
 	return 0;
 }
 
+/**
+ *
+ * @brief Create ibv_cq_ex by calling ibv_create_cq_ex
+ *
+ * @param[in] cq Pointer to the efa_cq.
+ * @param[in] attr Pointer to fi_cq_attr.
+ * @param[out] Return pointer to ibv_cq_ex if successful, otherwise NULL
+ */
+static inline struct ibv_cq_ex *rxr_ep_create_ibv_cq_ex(struct rxr_ep *ep, struct fi_cq_attr *attr) {
+	struct ibv_cq_init_attr_ex init_attr_ex = {
+		.cqe = attr->size ? attr->size : EFA_DEF_CQ_SIZE,
+		.cq_context = NULL,
+		.channel = NULL,
+		.comp_vector = 0,
+		/* EFA requires these values for wc_flags and comp_mask.
+		 * See `efa_create_cq_ex` in rdma-core.
+		 */
+		.wc_flags = IBV_WC_STANDARD_FLAGS,
+		.comp_mask = 0,
+	};
+
+	struct efa_domain *efa_domain = rxr_ep_domain(ep);
+	return ibv_create_cq_ex(efa_domain->device->ibv_ctx, &init_attr_ex);
+}
+
+static inline int rxr_ep_set_ibv_cq_ex(struct rxr_ep *ep, struct fi_cq_attr *attr) {
+	if (ep->ibv_cq_ex) {
```

Comment:
I think this function can be folded into `rxr_endpoint`.

Reply:
Not sure this check is useful.

I think an assertion is good enough.

Reply:
ACK